lib/zache.rb

```diff
@@ -124,6 +96,14 @@ def calc(key, lifetime)
     @hash[key][:value]
   end
 
+  def synchronized
```

Comment:
👍 I'm thinking about a name like `with_sync`, similar to many other libraries do, like `with_pool` for connection pools